subsys/bluetooth/host/gatt.c

```diff
@@ -5158,7 +5170,9 @@ void bt_gatt_connected(struct bt_conn *conn)
 
 #if defined(CONFIG_BT_GATT_CLIENT)
 	add_subscriptions(conn);
-
+#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU)
+	bt_gatt_exchange_mtu(conn, &gatt_exchange_params);
+#endif /* CONFIG_BT_GATT_AUTO_UPDATE_MTU */
```

Comment:
Do we need to check the role? 

The core spec says:

> The Client Rx MTU shall be greater than or equal to the default ATT_MTU.
> This request shall only be sent once during a connection by the client.

But who is the client?

For e.g. LE Audio both sides will act as a GATT server and GATT client, so both sides will attempt to do `bt_gatt_exchange_mtu` at the same time. We could consider adding a small delay for e.g. the peripheral. The same discussion was done for auto-connect EATT: https://github.com/zephyrproject-rtos/zephyr/pull/43112#pullrequestreview-902359990

Reply:
The both sides can be Client and Server at the same time. Even if both will initiate MTU Exchange at the same time there will no be a collision since the Tx and Rx MTU is the same (symmetrical).
Will take a look at the delay discossion :)

Reply:
@Thalley The discussion regarding EATT auto-connection is somehow different, as there might be collision when device allocated EATT Channels and sent request. When it received EATT connection request it check whether it has free channels, but it does not have because those might be in use (self-initiated request in progress) and return error. This will result in EATT Channels not being created.

Here, regarding MTU, there will not be collision, because MTU in ATT MTU Request has to be the same as the one that would be sent in ATT MTU Response.

Reply:
@MariuszSkamra But if both sides requests MTU exchange on connect (e.g. on 2 Zephyr devices), then both of them will fail, right?

But I see that we accept MTU requests even if `ATT_MTU_EXCHANGE_COMPLETE` or `ATT_MTU_EXCHANGE_IN_PROGRESS` is set, so it will currently not be a problem. 

If we have sent an MTU exchange to the peer, should we accept MTU exchange requests from that device too? i.e.

Central sends MTU request (sets `ATT_MTU_EXCHANGE_IN_PROGRESS`)
Peripheral sends MTU request (sets `ATT_MTU_EXCHANGE_IN_PROGRESS`)
Peripheral receives central's MTU request => Accept or reject since we are already waiting response?
Central receives peripheral's MTU request => Accept or reject since we are already waiting response?


There's also the other issue regarding multiple requests:
The spec states that a client shall only request MTU once. Should we reject multiple MTU requests from a client as a GATT server?

Reply:
> @MariuszSkamra But if both sides requests MTU exchange on connect (e.g. on 2 Zephyr devices), will both accept or reject?
Both accept. Why would Client reject MTU Exchange?
> 
> But I see that we accept MTU requests even if `ATT_MTU_EXCHANGE_COMPLETE` or `ATT_MTU_EXCHANGE_IN_PROGRESS` is set, so it will currently not be a problem.
> 
> If we have sent an MTU exchange to the peer, should we accept MTU exchange requests from that device too? i.e.
> 
> Central sends MTU request (sets `ATT_MTU_EXCHANGE_IN_PROGRESS`) Peripheral sends MTU request (sets `ATT_MTU_EXCHANGE_IN_PROGRESS`) Peripheral receives central's MTU request => Accept or reject since we are already waiting response? Central receives peripheral's MTU request => Accept or reject since we are already waiting response?
As the both sides can request MTU Exchange, the request shall not be rejected. The `ATT_MTU_EXCHANGE_IN_PROGRESS` and `ATT_MTU_EXCHANGE_COMPLETE` have been added to ensure the Client will not request MTU Exchange more than once. 
> 
> There's also the other issue regarding multiple requests: The spec states that a client shall only request MTU once. Should we reject multiple MTU requests from a client as a GATT server?

This would lead to potential interoperability issues, so I think it's better to sent the ATT MTU Response. We should be compliant with the Core but have a little sense of flexibility to what peer does.